samples/order/processor/processor.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A processor processes images in orders. It receives requests by
+// subscribing to the requests topic, and writes responses to the response
+// topic.
+//
+// At present, it just encodes input images into PNG format.
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"image"
+	_ "image/jpeg"
+	"image/png"
+	"log"
+	"time"
+
+	"gocloud.dev/blob"
+	_ "gocloud.dev/blob/fileblob"
+	"gocloud.dev/docstore"
+	_ "gocloud.dev/docstore/memdocstore"
+	"gocloud.dev/gcerrors"
+	"gocloud.dev/pubsub"
+	_ "gocloud.dev/pubsub/mempubsub"
+	"gocloud.dev/samples/order/internal/common"
+)
+
+var (
```

Comment:
Can we give default values for these flags that work with "local" implementations so that the example will work out of the box with no arguments?

Reply:
(and then you can remove the "missing" checks)

Reply:
Done, but I don't think it's satisfying because there's so much external setup needed.